cpu/stm32_common/periph/i2c.c

```diff
@@ -0,0 +1,463 @@
+/*
+ * Copyright (C) 2015 Jan Pohlmann <jan-pohlmann@gmx.de>
+ *               2017 we-sens.com
+ *               2018 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32l0
+ * @ingroup     drivers_periph_i2c
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation
+ *
+ * @note This implementation only implements the 7-bit addressing polling mode
+ * (for now interrupt mode is not available)
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @auhtor      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Jan Pohlmann <jan-pohlmann@gmx.de>
+ * @author      Aurélien Fillau <aurelien.fillau@we-sens.com>
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+
+#include "cpu.h"
+#include "mutex.h"
+
+#include "cpu_conf_stm32_common.h"
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define TICK_TIMEOUT    (0xFFFF)
+
+#if defined(CPU_FAM_STM32F0) || defined(CPU_FAM_STM32F3) || \
+    defined(CPU_FAM_STM32F7) || defined(CPU_FAM_STM32L0) || \
+    defined(CPU_FAM_STM32L4)
+
+#define I2C_IRQ_PRIO    (1)
+
+#define CLEAR_FLAG(dev)            dev->ICR |= I2C_ICR_NACKCF   | \
```

Comment:
we usually avoid macro, I would suggest to use a constant to only define the value, 
i.e. `#define CLEAR_FLAG (I2C_ICR_NACKCF | I2C_ICR_ARLOCF | I2C_ICR_BERRCF)`
and then use it `dev->ISR |= CLEAR_FLAG`

Reply:
Another method which will clear all w1c flags is
```c
dev->ISR = dev->ISR;
```